src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -113,23 +114,41 @@ public long GetMinimalEtagForReplication()
         {
             long minEtag = long.MaxValue;
 
+            DatabaseTopology topology;
             using (_server.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
-            using (ctx.OpenReadTransaction())
             {
-                var externals = _server.Cluster.ReadRawDatabaseRecord(ctx, Database.Name).ExternalReplications;
-                if (externals != null)
+                using (ctx.OpenReadTransaction())
                 {
-                    foreach (var external in externals)
+                    var dbRecord = _server.Cluster.ReadRawDatabaseRecord(ctx, Database.Name);
+                    topology = dbRecord.Topology;
+                    List <ExternalReplication> externals = dbRecord.ExternalReplications;
+                    if (externals != null)
                     {
-                        var state = GetExternalReplicationState(_server, Database.Name, external.TaskId, ctx);
-                        var myEtag = ChangeVectorUtils.GetEtagById(state.DestinationChangeVector, Database.DbBase64Id);
-                        minEtag = Math.Min(myEtag, minEtag);
+                        foreach (var external in externals)
+                        {
+                            var state = GetExternalReplicationState(_server, Database.Name, external.TaskId, ctx);
+                            var myEtag = ChangeVectorUtils.GetEtagById(state.DestinationChangeVector, Database.DbBase64Id);
+                            minEtag = Math.Min(myEtag, minEtag);
+                        }
                     }
                 }
             }
+            List<ReplicationNode> replicationNodes = new List<ReplicationNode>();
+
+            foreach (var node in topology.AllNodes)
+            {
+                if (node == _server.NodeTag)
+                    continue;
+                var internalReplication = new InternalReplication
+                {
+                    NodeTag = node,
+                    Url = _clusterTopology.GetUrlFromTag(node),
+                    Database = Database.Name
+                };
+                replicationNodes.Add(internalReplication);
+            }
 
-            var replicationNodes = Destinations?.ToList();
-            if (replicationNodes == null || replicationNodes.Count == 0)
+            if ( replicationNodes.Count == 0)
```

Comment:
for less code branching, I think we can remove this `if`

Reply:
done